### PR TITLE
feat: implement SignUp page component

### LINKED ISSUE #57: [frontend] implement SignUp(Register) page component
### Subject of this issue

Implement SignUp page **component**

### PR DESCRIPTION
**Pull request by dkim94**
### What is the issue this pull request related with?

resolved #57  

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Code builds correctly
- [ ] Created tests which fail without the change (if possible)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### Description
Please describe your pull request.

💔Thank you!

### CODE REVIEW

File: frontend/src/containers/SignUpContainer/SignUpForm.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React, { useCallback, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import SignUp from '../../components/SignUp/SignUp';
+import { register, changeField, initalizeForm } from '../../modules/auth/auth';
+
+function SignUpForm({ history }) {
+  const dispatch = useDispatch();
+  const form = useSelector(({ auth }) => ({
```
Comment (zeroFruit) [inline frontend/src/containers/SignUpContainer/SignUpForm.js]: To fix `form.form` to `form`, change `const form` into `const { form }`

Please reference this link: https://wesbos.com/destructuring-objects/

File: frontend/src/components/SignUp/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+SignUp.propTypes = {
+  form: PropTypes.object.isRequired,
+  onChange: PropTypes.func.isRequired,
+  onClick: PropTypes.func.isRequired,
+};
+
+function SignUp({ form, onChange, onClick }) {
+  return (
+    <SignUpBlock>
+      <h3>Join Ya-Ta!</h3>
```
Comment (zeroFruit) [inline frontend/src/components/SignUp/SignUp.js]: please change all heading tag `<h1>`, `<h2>`, ... into `<Typography variant="h1">` ... 

you need to `import { Typography } from '@material-ui/core';`

File: frontend/src/components/SignUp/SignUp.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          value={form.passwordConfirmation}
+        />
+        <Typography variant="h4">Vehicle Info (for Driver users)</Typography>
+        <Typography variant="h5">Car Type</Typography>
+        <input
+          type="text"
+          id="car-type-input"
+          name="carType"
+          onChange={onChange}
+          value={form.vehicleInfo.carType}
```
Comment (dkim94) [inline frontend/src/components/SignUp/SignUp.js]: Is it okay to put value like this?

File: backend/user_service/user/app/user_application_service.py [hunk 4]
```diff
@@ -16,24 +18,44 @@
...
+
+
+class UserSerializer(serializers.ModelSerializer):
+
+    vehicle = VehicleSerializer(many=False, read_only=True)
+
+    class Meta:
+        model = User
+        fields = ('id', 'email', 'user_type', 'vehicle')
+        read_only_fields = ('id', 'email', 'user_type')
```
Comment (zeroFruit) [inline backend/user_service/user/app/user_application_service.py]: Move these serializers into user_service domain

**Pull request merged**